interview-questions.md

```diff
@@ -9,62 +9,90 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Your answer:
 
+    A function is logic code that can be reused throughout your program to save time and make your life easier.
+
   Researched answer:
 
+    A function is a set of instructions for the computer to execute. It can take some (optional) inputs and adjust the data and return (optional) outputs when called on to save time when doing repetitive tasks.
+
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
 
+    .map() iterates over every value in an array and performs some action to it. .filter() keeps any value in an array that meets a conditional.
+
   Researched answer:
 
+    Both return new arrays. .map() performs a function on every value in an array and returns the new adjusted array. .filter() checks every value in an array for a true statement and filters out the false, forming a new array with the filtered values.
+
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
 
+    console.log() is only seen in the terminal or console, it's not generally visible to the user. Return is used within a function to indicate whatever input you want to give back to the user. Functions must have a return.  
+
   Researched answer:
 
+    console.log() displays a parameter in the console window. When a return statement is used, a function will stop executing and return the specified value.
+
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
 
+    An argument is any data the user inputs into the function.
```

Comment:
Yup! But, not just the user! A different function could be passing information into the argument as well. So other parts of your program can also take advantage of the argument.